src/parser/keyword.c

```diff
@@ -527,9 +527,28 @@ static KEYWORD_RECORD *pt_find_keyword (const char *text);
 static int keyword_cmp (const void *k1, const void *k2);
 
 
+/* The GET_KEYWORD_HASH_VALUE() macro is the definition of the djb2 algorithm as a macro.
+ * Refer to the string_hash() function implemented in the libcubmemc.c file.
+ */
+#define GET_KEYWORD_HASH_VALUE(h,s) \
+  do { \
+      unsigned char* p = (unsigned char*)(s); \
+      for((h) = 5381;  *p; p++ ) \
+        { \
+             (h) = (((h) << 5) + (h)) + *p; /* hash * 33 + c */ \
```

Comment:
When using the djb2 hash function, use the unsigned long type.

However, the hash_value variable of the KEYWORD_RECORD struct uses an unsigned short type.

Is there any possibility of overflow while repeating h * 33?

Reply:
I looked for content that overflows often occur when using a hash function. But isn't the unsigned short type small enough to cause frequent collisions?

Since the keyword is not long, is it okay if it is small?

Reply:
The hash value does not matter if overflow occurs.
Even if overflow occurs, all you need to do is show an even distribution.